docs/usage/docs.md

```diff
@@ -2,6 +2,231 @@
+- We use a handful of custom [template plugins](#template-plugins)
+
+## Markdown
+Whenever possible, please write your documents in [Markdown] rather than HTML.
+The syntax is easier to read and write, and allows us to enhance the content
+programmatically with [remark](#remark).
+
+### Frontmatter
+The term [frontmatter][] (sometimes written as "front matter" or "front
+matter") was popularized by a static site generator called [Jekyll], which
+heavily influenced Eleventy's conventions. Like Jekyll, Eleventy allows us to
+embed metadata with frontmatter written in [YAML] at the top of each page.
+YAML frontmatter is delimited by an opening and closing `---` starting on the
+first line of the file:
+
+```yaml
+---
+key: value
+list: [a, b, c]
+# comments look like this
+objects:
+  can:
+    nest: deeply
+---
+
+Everything from here on out is treated as Markdown.
+```
+
+The only **required** piece of metadata is the page's title:
+
+```yaml
+---
+title: Title goes here
+---
+```
+
+You may add whatever data you want, and you can even reference it in your
+document with [Nunjucks](#nunjucks) in your markdown. For instance, to generate
+a list of links with predictable URLs you could do something like this:
+
+
+```md
+---
+fruits:
+  - apple
+  - strawberry
+  - pear
+---
+
+## Fruits
+{% raw %}{% for fruit in fruits %}
+- [{{ fruit }}](https://google.com/search?q={{ fruit }})
+{% endfor %}{% endraw %}
+```
+
+This becomes really powerful when generating [code examples](#code-examples)
+from data, e.g. in an HTML [code block]:
+
+    ```html
+    <ul>
+    {% raw %}{% for key, value in tokens.colors %}
+      <li>
+        <b>{{ key }}</b>: <code>{{ value }}</code>
+        <span class="text-{{ key }}">text-{{ key }}</span> /
+        <span class="bg-{{ key }}">bg-{{ key }}
+      </li>
+    {% endfor %}{% endraw %}
+    </ul>
+    ```
+
+## Remark
+[Remark] is a tool for parsing and transforming Markdown with plugins. Support
+for Markdown is [built into][eleventy markdown] Eleventy, but we use [an
+Eleventy plugin](https://npm.im/@fec/eleventy-plugin-remark) to configure
+Remark as Eleventy's Markdown "engine" so that we can process it with Remark
+plugins. Remark has three distinct phases to be aware of:
+
+- The **parse** phase takes raw text from the `.md` documents and creates a
+  [tree structure][AST] in which each distinct piece of the document &mdash; a
+  heading, list item, link, or other tag &mdash; is represented as a **node**
+  in the tree.
+
+- The **transform** phase applies plugins that walk the tree and add, remove,
+  or update nodes.
+
+- The **stringify** phase walks the tree to generate a string of HTML. This
+  phase uses [Rehype].
+
+⚠️ It's important to remember that Markdown documents are processed with Nunjucks
+**before** they're parsed with Remark. This can cause problems if your template
+instructions produce malformed Markdown, but you can always drop into HTML if
+need be.
```

Comment:
do we have an in-house convention for warnings/notes/attention-grabbers? could be nice to include if so, but no big